app/src/sharedTest/java/org/oppia/android/app/settings/profile/ProfileEditFragmentTest.kt

```diff
@@ -158,6 +158,25 @@ class ProfileEditFragmentTest {
     }
   }
 
+  @Test
+  fun testProfileEdit_startWithUserProfile_clickProfileDeletionButton_deleteSnackbarIsVisible() {
+    launchFragmentTestActivity(internalProfileId = 1).use {
+      onView(withId(R.id.profile_delete_button)).perform(click())
+      onView(withId(android.R.id.button1)).perform(click())
+      onView(withId(com.google.android.material.R.id.snackbar_text))
+        .check(matches(withText(R.string.profile_edit_delete_success)))
+    }
+  }
+
+  @Test
+  fun testProfileEdit_startWithUserProfile_AfterDeleteSnackbar_profileActivityIsShownAgain() {
+    launchFragmentTestActivity(internalProfileId = 1).use {
+      onView(withId(R.id.profile_delete_button)).perform(click())
+      onView(withId(android.R.id.button1)).perform(click())
+      intended(hasComponent(ProfileListActivity::class.java.name))
+    }
```

Comment:
This will not work for the scenario you are trying to test. You can build on my suggestion in the test above:

```suggestion
 onView(withId(R.id.profile_delete_button)).perform(click())
      testCoroutineDispatchers.runCurrent()
      onView(withText(R.string.profile_edit_delete))
        .inRoot(isDialog())
        .perform(click())
      testCoroutineDispatchers.runCurrent()
// You will need to click on the snackbar dismiss button
onView(allOf(withId(android.support.design.R.id.snackbar_action)))
    .perform(click())
  testCoroutineDispatchers.runCurrent()
  intended(hasComponent(ProfileListActivity::class.java.name))
```